common/src/gas_provider/node.rs

```diff
@@ -105,6 +144,38 @@ impl<ExternalId: Clone, Id: Clone + Copy, Balance: Zero + Clone + Copy>
     }
 
     pub fn refs(&self) -> u32 {
-        self.spec_refs.saturating_add(self.unspec_refs)
+        self.spec_refs().saturating_add(self.unspec_refs())
+    }
+
+    pub fn spec_refs(&self) -> u32 {
+        match self.inner {
+            GasNodeType::External { refs, .. } | GasNodeType::SpecifiedLocal { refs, .. } => {
+                refs.spec_refs
+            }
+            _ => 0,
+        }
```

Comment:
```suggestion
        use GasNodeType::{External, SpecifiedLocal};
        
        match self.inner {
            External { refs, .. } | SpecifiedLocal { refs, .. } => {
                refs.spec_refs
            }
            _ => 0,
        }
```

May be this var is more readable? And in other places

Reply:
please take a look at last changes ([64afdb6](https://github.com/gear-tech/gear/pull/1225/commits/64afdb68d5f0e878e199d15f19e4965ec8270ebe) and tell me whether you think changing it the way you suggested is still necessary.